app/controllers/api/v1/orders_controller.rb

```diff
@@ -9,6 +9,20 @@ def cart
     render json: get_cart
   end
 
+  def get_placed
+    render json: get_placed_order
+  end
+
+  def change_state
+    @order = Order.find(params[:id])
+
+    if @order.make_placed
```

Comment:
this should look for ```params[:state]``` that's passed. so then (after validating that the state passed is part of the possible order states) can do ```@order.send("make_#(state)")``` so you can move an order to any state.